src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Directives.cs

```diff
@@ -27,14 +28,27 @@ private void SetupDirectives()
             MapDirectives(LayoutDirective, SyntaxConstants.CSharp.LayoutKeyword);
         }
 
+        protected virtual void TagHelperPrefixDirective()
+        {
+            TagHelperDirective(
```

Comment:
:-1: The format of all tag helper directives is the same, from a parsing perspective they're nearly identical, making a whole other method where the largest change is the CodeGenerator isn't worth it.